### PR TITLE
Stellar: Transaction details

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This imports some subcomponents from wallet/transaction/index.js -- perhaps it makes sense to split those out into their own JS files, let me know what you think.

### CODE REVIEW

File: shared/wallets/transaction/index.js [hunk 1]
```diff
@@ -184,31 +222,42 @@ export const Transaction = (props: Props) => {
...
+            pending={pending}
+            yourRole={props.yourRole}
+            counterparty={props.counterparty}
+            counterpartyType={props.counterpartyType}
+            amountUser={props.amountUser}
+          />
+          {// TODO: Consolidate memo display code below with
+          // chat/conversation/messages/wallet-payment/index.js.
+          showMemo && (
+            <Box2 direction="horizontal" gap="small" fullWidth={true}>
```
Comment (buoyad) [inline shared/wallets/transaction/index.js]: This should maybe have a marginTop, it looks too close to the `Detail`

File: shared/wallets/transaction-details/index.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  counterparty: string,
+  counterpartyMeta?: string,
+  counterpartyType: CounterpartyType,
+  isYou: boolean,
+  you: string,
+  yourRole: Role,
+|}
+
+const Counterparty = (props: CounterpartyProps) =>
+  props.isYou ? (
```
Comment (buoyad) [inline shared/wallets/transaction-details/index.js]: nested ternary makes this hard to read, replace with `if`?

File: shared/wallets/transaction-details/index.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      />
+      <Box2
+        direction="vertical"
+        fullWidth={true}
+        style={{justifyContent: 'center', marginLeft: globalMargins.small}}
+      >
+        <CounterpartyText
+          counterparty={props.counterparty}
+          counterpartyType={props.counterpartyType}
+          large={true}
```
Comment (buoyad) [inline shared/wallets/transaction-details/index.js]: same here

File: shared/wallets/transaction-details/index.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
+// @flow
+import * as React from 'react'
+import {Box2, Divider, Icon, NameWithIcon, Text} from '../../common-adapters'
+import {globalColors, globalMargins, styleSheetCreate} from '../../styles'
+import Transaction, {CounterpartyIcon, CounterpartyText, Timestamp} from '../transaction'
+
+type Role = 'sender' | 'receiver'
+type CounterpartyType = 'keybaseUser' | 'stellarPublicKey' | 'wallet'
+
+export type Props = {|
```
Comment (buoyad) [inline shared/wallets/transaction-details/index.js]: can these all be either optional or nullable?

File: shared/wallets/transaction-details/index.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        yourRole={props.yourRole}
+      />
+    </Box2>
+
+    <Box2 direction="vertical" gap="xxtiny" fullWidth={true}>
+      <Text type="BodySmallSemibold">Status:</Text>
+      <Box2 direction="horizontal" fullHeight={true} fullWidth={true} style={{alignItems: 'center'}}>
+        <Icon
+          color={props.timestamp ? globalColors.green2 : globalColors.black}
+          size="16"
```
Comment (buoyad) [inline shared/wallets/transaction-details/index.js]: `fontSize`

**Comment by cjb** [comment]
@buoyad Thanks! Made all of these changes.

File: shared/wallets/transaction-details/index.js [hunk 6]
```diff
@@ -33,38 +33,45 @@ type CounterpartyProps = {|
...
+        metaOne={props.counterpartyMeta}
       />
-      <Box2
-        direction="vertical"
-        fullWidth={true}
-        style={{justifyContent: 'center', marginLeft: globalMargins.small}}
-      >
-        <CounterpartyText
+    )
+  } else {
```
Comment (buoyad) [inline shared/wallets/transaction-details/index.js]: unneeded `else`

**Comment by cjb** [comment]
Thanks! Did those too.

**Pull request merged**